### PR TITLE
fix: color mode

### PR DESCRIPTION
**Pull request by nhsz**
This PR adds `initialColorMode` to `ColorModeScript` to save color on localStorage

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ const App = ({ Component, pageProps }: AppPropsWithLayout) => {
           }
         `}
       </style>
-      <ChakraProvider theme={theme}>
+      <ChakraProvider theme={theme} colorModeManager={localStorageManager}>
```
Comment (TylerAPfledderer) [inline src/pages/_app.tsx]: `localStorageManager` should not need to be imported as it is used by default for `colorModeManager`.
[colorModeManager under the hood](https://github.com/chakra-ui/chakra-ui/blob/b6befea762b44f923af42792473963215dc50ed1/packages/components/color-mode/src/color-mode-provider.tsx#L40)

Is there something that is not working without declaring it here?

**Review by TylerAPfledderer — CHANGES_REQUESTED** [review]
Hey @nhsz! One quick thought from me. 😄

**Comment by wackerow** [comment]
@TylerAPfledderer We're trying to patch the flash when a user is in dark mode and refreshes the page: https://deploy-preview-105--ethereum-org-fork.netlify.app/en/about

If you go there, make sure you're in dark mode and refresh, you'll see the main background flashes white before fading to the desired color. Any idea why this behaves this way?

**Comment by wackerow** [comment]
Also noting that it's a nice-to-have, since the existing site already does this... don't want to cause too much scope creep, but if you know of a quick fix

**Comment by TylerAPfledderer** [comment]
> @TylerAPfledderer We're trying to patch the flash when a user is in dark mode and refreshes the page: https://deploy-preview-105--ethereum-org-fork.netlify.app/en/about
> 
> If you go there, make sure you're in dark mode and refresh, you'll see the main background flashes white before fading to the desired color. Any idea why this behaves this way?

Now that the site is going into NextJS, it's much easier to access cookies compared to Gatsby.

Have you considered extending the use of [colorModeManager](https://chakra-ui.com/docs/styled-system/color-mode#add-colormodemanager-optional-for-ssr) to use a cookie? Or would a cookie not be possible in this project's use?

File: src/pages/_app.tsx (hunk 1 above)
Comment (nhsz) [reply src/pages/_app.tsx]: We had this same issue in other NextJS sites with Chakra and solved the same way, seems for NextJS is required to set `colorModeManager` (https://chakra-ui.com/docs/styled-system/color-mode#add-colormodemanager-optional-for-ssr)

**Comment by nhsz** [comment]
> > @TylerAPfledderer We're trying to patch the flash when a user is in dark mode and refreshes the page: https://deploy-preview-105--ethereum-org-fork.netlify.app/en/about
> > If you go there, make sure you're in dark mode and refresh, you'll see the main background flashes white before fading to the desired color. Any idea why this behaves this way?
> 
> Now that the site is going into NextJS, it's much easier to access cookies compared to Gatsby.
> 
> Have you considered extending the use of [colorModeManager](https://chakra-ui.com/docs/styled-system/color-mode#add-colormodemanager-optional-for-ssr) to use a cookie? Or would a cookie not be possible in this project's use?

what would be the advantage vs using localStorage?

**Comment by TylerAPfledderer** [comment]
> > > @TylerAPfledderer We're trying to patch the flash when a user is in dark mode and refreshes the page: https://deploy-preview-105--ethereum-org-fork.netlify.app/en/about
> > > If you go there, make sure you're in dark mode and refresh, you'll see the main background flashes white before fading to the desired color. Any idea why this behaves this way?
> > 
> > 
> > Now that the site is going into NextJS, it's much easier to access cookies compared to Gatsby.
> > Have you considered extending the use of [colorModeManager](https://chakra-ui.com/docs/styled-system/color-mode#add-colormodemanager-optional-for-ssr) to use a cookie? Or would a cookie not be possible in this project's use?
> 
> what would be the advantage vs using localStorage?

If I understand correctly, a cookie can be read in the server so it's known ahead of time what the value of the cookie is before painting the page.

However, I am going to try and check with the Chakra team on this to see what are the current known symptoms of the flashing, because the preview deploy of this PR still has it.

**Comment by TylerAPfledderer** [comment]
@wackerow @nhsz it's the `mode()` method used for the [body tag background style](https://github.com/ethereum/ethereum-org-fork/blob/729426d1094eb78ff4b9fe9f6289b2a8886d27a2/src/%40chakra-ui/styles.ts#L17). This needs to be flipped to the semantic `background.base`.

I don't think the SSR support is needed here, and recommend removing the declaration of the `colorModeManager` prop with `localStorageManager` as this is already done by default and therefore redundant. (Removing this locally had no negative impact after replacing `mode()`.)

**Comment by wackerow** [comment]
Okay thanks @TylerAPfledderer, will check this out when I get time

**Comment by pettinarip** [comment]
Nice @TylerAPfledderer, thanks for investigating.

~~So, given that this appears to be a bug coming from the gatsby repo, we should fix it there and bring it here in our regular updates.~~

~~@wackerow if you can confirm this works well, I'd suggest to close this PR and create a fix in the gatsby repo.~~


====
**edit**
Confirming that this works well (by changing the body color to `background.base`) but it only works if we declare the `colorModeManager`. I did try **not** declaring it but I had the same flashing issue again.

I pushed a new commit with the patch. Once we've tested the preview deploy, I think we can merge it (I'll create a PR in the Gatsby repo as well).

=====
**edit2**
Double-checked this again and confirming that removing the `localStorageManager` also worked well, as it was supposed.

**Review by pettinarip — APPROVED** [review]

**Pull request merged**